akka-http-core/src/main/scala/akka/http/impl/engine/client/OutgoingConnectionBlueprint.scala

```diff
@@ -352,4 +357,8 @@ private[http] object OutgoingConnectionBlueprint {
       override def preStart(): Unit = getNextMethod()
     }
   }
+
+  class UnexpectedConnectionClosureException(outstandingResponses: Int)
+    extends RuntimeException(s"The http server closed the connection unexpectedly before delivering responses for $outstandingResponses outstanding requests")
```

Comment:
I started adding more info but I then figured that it isn't possible to provide all possible causes of this comprehensively. I wonder if we shouldn't create more detailed explanations for some errors in the documentation instead which might also be easier to maintain?